### PR TITLE
feat: clean QuickSight temp resources created for preview

### PR DESCRIPTION
**Pull request by llmin**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

(describe what this merge request does)

## Implementation highlights

(describe how the merge request does for feature changes, share the RFC link if it has)

## Test checklist

- [x] add new test cases
- [ ] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [ ] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

**Review by zxkane — COMMENTED** [review]
resolve conflicts locally

File: src/control-plane/backend/lambda/api/service/reporting.ts [hunk 1]
```diff
@@ -117,7 +117,10 @@ export class ReportingServ {
       const dashboardCreateParameters = query.dashboardCreateParameters as DashboardCreateParameters;
 
       //construct parameters to build sql
-      const viewName = query.viewName;
+      let viewName = query.viewName;
+      if (query.action === ExploreRequestAction.PREVIEW) {
```
Comment (zxkane) [inline src/control-plane/backend/lambda/api/service/reporting.ts:121]: create a shared method for creating temp view name

File: src/control-plane/backend/lambda/api/service/reporting.ts [hunk 2]
```diff
@@ -882,5 +894,85 @@ export class ReportingServ {
...
+        ...aws_sdk_client_common_config,
+      });
+
+      const dashBoards = await quickSight.listDashboards({
+        AwsAccountId: awsAccountId,
+      });
+
+      if (dashBoards.DashboardSummaryList) {
+        for (const [_index, dashboard] of dashBoards.DashboardSummaryList.entries()) {
+          if (dashboard.Name?.startsWith('_tmp_') && (new Date().getTime() - dashboard.CreatedTime!.getTime()) > 60*60*1000) {
```
Comment (zxkane) [inline src/control-plane/backend/lambda/api/service/reporting.ts:919]: extract `_tmp_` as an internal private constant

**Review by zxkane — CHANGES_REQUESTED** [review]

File: src/control-plane/backend/lambda/api/service/reporting.ts (hunk 1 above)
Comment (llmin) [reply src/control-plane/backend/lambda/api/service/reporting.ts]: fixed

File: src/control-plane/backend/lambda/api/service/reporting.ts (hunk 2 above)
Comment (llmin) [reply src/control-plane/backend/lambda/api/service/reporting.ts:865]: fixed

File: src/control-plane/backend/lambda/api/router/reporting.ts [hunk 3]
```diff
@@ -56,6 +56,12 @@ reporting_project.post(
     return reportingServ.warmup(req, res, next);
   });
 
+reporting_project.post(
```
Comment (zxkane) [inline src/control-plane/backend/lambda/api/router/reporting.ts:59]: @tyyzqmf should we add this route to middleware to grant the permission for the role analyst?

Comment (tyyzqmf) [reply src/control-plane/backend/lambda/api/router/reporting.ts:59]: The rule already includes this route: `routerRoles.set('ALL /api/metadata/*', [IUserRole.ADMIN, IUserRole.ANALYST])`

**Review by zxkane — APPROVED** [review]

**Pull request merged**